### PR TITLE
fix: Move the code from bloomstack_core to erpnext for work order

### PR DESCRIPTION
**Pull request by shruti2323**
Fix Description: Moved the code from bloomstack_core to erpnext.
Asana Link: https://app.asana.com/0/1200073273462988/1200278560808264

**PR of bloomstack core fix: https://github.com/Bloomstack/bloomstack_core/pull/607**

### CODE REVIEW

File: erpnext/manufacturing/doctype/work_order/work_order.js [hunk 1]
```diff
@@ -66,6 +66,40 @@ frappe.ui.form.on("Work Order", {
 		erpnext.queries.setup_queries(frm, "Warehouse", function() {
 			return erpnext.queries.warehouse(frm.doc);
 		});
+		
+		// get meta information of doctype Work Order Operation
```
Comment (shruti2323) [inline erpnext/manufacturing/doctype/work_order/work_order.js:70]: Code shuffled from app bloomstack_core work_order.js to erpnext.

File: erpnext/manufacturing/doctype/work_order/work_order.json [hunk 2]
```diff
@@ -13,6 +13,7 @@
   "item_name",
   "image",
   "bom_no",
+  "title",
```
Comment (shruti2323) [inline erpnext/manufacturing/doctype/work_order/work_order.json:16]: Below is the Image attached for Work Order custom form fields and Doctype field difference, where "Title" was the custom field which added to Work Order doctype fields.

https://user-images.githubusercontent.com/80836439/122998481-ba74bf80-d3ca-11eb-9d81-12e0a73d0470.mp4

Work_order_before
Work_order_before1

File: erpnext/manufacturing/doctype/work_order/work_order.js [hunk 3]
```diff
@@ -66,6 +66,40 @@ frappe.ui.form.on("Work Order", {
...
+				const escaped_name = encodeURIComponent(value);
+
+				return repl('<a class="indicator %(color)s" href="#Form/%(doctype)s/%(name)s">%(label)s</a>', {
+					color: color,
+					doctype: df.options,
+					name: escaped_name,
+					label: label
+				});
+			};
 	},
```
Comment (vishdha) [inline erpnext/manufacturing/doctype/work_order/work_order.js:103]:
```suggestion
	
```
you directly add this indicator color and condition in work_order_list.js

**Review by vishdha — CHANGES_REQUESTED** [review]

File: erpnext/manufacturing/doctype/work_order/work_order_list.js [hunk 4]
```diff
@@ -17,3 +17,40 @@ frappe.listview_settings['Work Order'] = {
 		}
 	}
 };
+
+frappe.ui.form.on("Work Order", {
```
Comment (shruti2323) [inline erpnext/manufacturing/doctype/work_order/work_order_list.js]: Suggested Change: you directly add this indicator color and condition in work_order_list.js

**Review by vishdha — CHANGES_REQUESTED** [review]
@shruti2323  can you update the pr. request changes is not here. work_order_list.js missing and outdated code still in pr

**Comment by shruti2323** [comment]
> @shruti2323 can you update the pr. request changes is not here. work_order_list.js missing and outdated code still in pr

**Comment by shruti2323** [comment]
> @shruti2323 can you update the pr. request changes is not here. work_order_list.js missing and outdated code still in pr



> @shruti2323 can you update the pr. request changes is not here. work_order_list.js missing and outdated code still in pr
Hello @vishdha, 
since the "operation" is a child table for the work order.
adding a logic inside work_order_list.js would not suffice.
frappe.listview_settings would work for all DocTypes except Child Tables and Single DocTypes.
and we are adding this logic under work_order.js to trigger once the form is getting created.

As per your suggestion, moving logic to work_order_list will not work because the return would trigger to doc, not to the child table.

image

**Pull request closed without merging**